jerry-core/ecma/operations/ecma-regexp-object.c

```diff
@@ -286,8 +279,24 @@ ecma_op_regexp_initialize (ecma_object_t *regexp_obj_p, /**< RegExp object */
   JERRY_UNUSED (flags);
 #endif /* ENABLED (JERRY_ESNEXT) */
 
+  ecma_value_t old_bc = ext_obj_p->u.class_prop.u.value;
+
   ext_obj_p->u.class_prop.class_id = LIT_MAGIC_STRING_REGEXP_UL;
   ECMA_SET_INTERNAL_VALUE_POINTER (ext_obj_p->u.class_prop.u.value, bc_p);
+
+  ecma_value_t set_last_index = ecma_builtin_helper_def_prop (regexp_obj_p,
```

Comment:
I'd rather not move this into regexp_initialize, the only place where this can throw is in the compile method. Cant we just move the def_prop call in the compile method down into the final `if (!ECMA_IS_VALUE_ERROR (ret_value))` block?